File: TopInterview150/Math/Pow/solution.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if (n == 0) return 1;

        long exp = n;
        if (exp < 0) {
            exp = -exp;
            x = 1/x;
        }
        
        double result = 1.0;
        while (exp > 0)
        {
            if (exp % 2 == 1)
            {
                result *= x;
            }
            x*=x;
            exp /= 2;
        }
        return result;
    }
};

File: TopInterview150/Math/Pow/solution.py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        if n == 0:
            return 1
        if n < 0:
            n = n * -1
            x = 1/x

        result = 1
        while n > 0:
            if n % 2 == 1:
                result = result * x
            x = x*x
            n = n // 2

        return result